prov/efa/src/efa_fabric.c

```diff
@@ -479,13 +492,40 @@ static int efa_get_gdr_support(char *device_name)
 		goto out;
 	}
 
-	ret = fi_read_file(get_sysfs_path(), gdr_path, &gdr_support, sizeof(char));
-	if (ret < 0)
+	ret = fi_read_file(get_sysfs_path(), gdr_path,
+			   gdr_support_str, MAX_GDR_SUPPORT_STRLEN-1);
+	if (ret < 0) {
+		ret = 0;
+		*gdr_support = 0;
+		goto out;
+	}
+
+	if (ret == 0) {
+		EFA_WARN(FI_LOG_FABRIC, "Sysfs file %s is empty\n", gdr_path);
+		ret = -FI_EINVAL;
+		goto out;
+	}
+
+	gdr_support_str[ret] = '\0';
+	*gdr_support = strtol(gdr_support_str, &endptr, 10);
+	if ((endptr - gdr_support_str) != 1) {
```

Comment:
strlen() here instead? Or check ret from `fi_read_file`? Not sure why we're doing pointer math here.

Reply:
This is checking ```gdr_support_ptr``` has the right content, e.g. instead of `0` and `1`, it contain `a`, `b`, etc.
strlen will not help in that case.

Reply:
I see, thanks, read the manpage and this makes sense now.